core/src/main/java/org/opensearch/sql/executor/OpenSearchTypeSystem.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.executor;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rel.type.RelDataTypeSystemImpl;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+public class OpenSearchTypeSystem extends RelDataTypeSystemImpl {
+  public static final RelDataTypeSystem INSTANCE = new OpenSearchTypeSystem();
+
+  private OpenSearchTypeSystem() {}
+
+  public RelDataType deriveAvgAggType(RelDataTypeFactory typeFactory, RelDataType argumentType) {
+    switch (argumentType.getSqlTypeName()) {
+      case INTEGER:
+      case BIGINT:
+        return typeFactory.createSqlType(SqlTypeName.DOUBLE);
```

Comment:
is it a bug in Calcite? if the argumentType is INTEGER, deriveAvgAggType is INTEGER?

Reply:
I think so,  for example table1:
| id |
|------|
| 1    |
| 2    |
| 3    |
| 4    |

`select avg(id) from table1` **returns 2 instead of 2.5**.

Not sure it is a bug or by designed. From semantic purpose, query avg on an `id` column is semantic wrong. The column which could be applied avg should be designed as decimal number. Will open a Calcite issue for checking.